components/calendar/src/calendar.rs

```diff
@@ -100,6 +100,12 @@ pub trait Calendar: crate::cal::scaffold::UnstableSealed {
     /// If left empty, any algorithm will parse successfully.
     fn calendar_algorithm(&self) -> Option<crate::preferences::CalendarAlgorithm>;
 
+    /// Returns the [`AnyCalendarKind`](crate::AnyCalendarKind) associated with this calendar,
+    /// if any.
+    ///
+    /// Can be used for runtime calendar checking.
+    fn calendar_kind(&self) -> Option<crate::AnyCalendarKind>;
```

Comment:
thought: I definitely feel like we can avoid this method since we have IntoAnyCalendar. But maybe I'm wrong here?

Reply:
I don't want to force `format_unchecked` to require `IntoAnyCalendar`.

Reply:
Hm, fair enough.

Reply:
I had just removed this in https://github.com/unicode-org/icu4x/pull/6456